modules/replicator/src/main/java/org/apache/ignite/internal/replicator/message/ReplicaResponse.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.replicator.message;
+
+import java.util.UUID;
+import org.apache.ignite.network.NetworkMessage;
+
+/**
+ * Replica response interface.
+ * TODO:IGNITE-17258 Add a specific response type for a replica listener. (@Transferable(ReplicaMessageGroup.TYPE_RESPONSE))
+ */
+public interface ReplicaResponse extends NetworkMessage {
+
+    /**
+     * Operation id in the replica with which the response is sent.
+     *
+     * @return Operation id.
+     */
+    UUID operationId();
```

Comment:
As far as I understand there are three types of responses 
-  Instant responses, e.g. read responses or write responses that already finish its replication. For such responses operationId doesn't have any sense.
- Delayed responses, e.g. write responses that didn't finish its replication yet. For such responses operation id has sense.
- Mixed responses, e.g. getAndRemove().

Reply:
Divided responses to three types as you proposed.